packages/jest-console/src/matchers.ts

```diff
@@ -51,22 +66,21 @@ const createSpyInfo = ( spy, matcherName, methodName, expected ) => {
 };
 
 const createToHaveBeenCalledMatcher =
-	( matcherName, methodName ) => ( received ) => {
-		const spy = received[ methodName ];
+	( matcherName: string, methodName: string ) => ( received: any ) => {
+		const spy = received[ methodName ] as Mock;
```

Comment:
We can get rid of `any` type and the assertion
```suggestion
	( matcherName: string, methodName: string ) =>
	( received: Record< string, Mock > ) => {
		const spy = received[ methodName ];
```